api/v1alpha1/envoygateway_helpers.go

```diff
@@ -96,6 +96,14 @@ func (e *EnvoyGateway) NamespaceMode() bool {
 		len(e.Provider.Kubernetes.Watch.Namespaces) > 0
 }
 
+// GatewayNamespaceMode returns true if controller uses gateway namespace mode for infra deployments.
+func (e *EnvoyGateway) GatewayNamespaceMode() bool {
```

Comment:
Would prefer a clearer name to make it easier to read. For example: `IsInfraDeployedInGatewayNamespace`

Reply:
I am okay with changing the name, however, I would like to keep a shorter name, just like `GatewayNamespaceMode`